crates/bws/src/command/run.rs

```diff
@@ -0,0 +1,133 @@
+use std::{
+    collections::{HashMap, HashSet},
+    io::{IsTerminal, Read},
+    process,
+};
+
+use bitwarden::{
+    secrets_manager::{
+        secrets::{SecretIdentifiersByProjectRequest, SecretIdentifiersRequest, SecretsGetRequest},
+        ClientSecretsExt,
+    },
+    Client,
+};
+use color_eyre::eyre::{bail, Result};
+use uuid::Uuid;
+use which::which;
+
+use crate::{
+    util::{is_valid_posix_name, uuid_to_posix},
+    ACCESS_TOKEN_KEY_VAR_NAME,
+};
+
+pub(crate) async fn run(
+    client: Client,
+    organization_id: Uuid,
+    project_id: Option<Uuid>,
+    uuids_as_keynames: bool,
+    no_inherit_env: bool,
+    shell: Option<String>,
+    command: Vec<String>,
+) -> Result<i32> {
+    let is_windows = std::env::consts::OS == "windows";
+
+    let shell = match is_windows {
+        true => shell.unwrap_or_else(|| "powershell".to_string()),
+        false => shell.unwrap_or_else(|| "sh".to_string()),
+    };
```

Comment:
I would generally write this as

```suggestion
    let shell = shell.unwrap_or_else(|| {
        if is_windows {
            "powershell".to_string()
        } else {
            "sh".to_string()
        }
    });
```

Reply:
Applied in 75e3d8b918e5c7adc9f5e0d99975b3866dc5753d.